From ce26fa998c4966e8a34d9e3be569d31636a12cf7 Mon Sep 17 00:00:00 2001
From: WingMan Kwok <w-kwok2@ti.com>
Date: Tue, 13 Sep 2016 14:45:56 -0400
Subject: [PATCH 37/58] ARM: dts: keystone: Add bindings for 10gbe MCU firmware

This patch adds the devicetree bindings for 10gbe
MCU firmware. These bindings are disabled by default.
When 10gbe MCU firmware is used for configuring the
10gbe SerDes, these bindings need to be explicitly enabled.

Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
Signed-off-by: Michael Scherban <m-scherban@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2e-netcp.dtsi  | 26 ++++++++++++++++++++++++++
 arch/arm/boot/dts/keystone-k2hk-netcp.dtsi | 26 ++++++++++++++++++++++++++
 2 files changed, 52 insertions(+)

diff --git a/arch/arm/boot/dts/keystone-k2e-netcp.dtsi b/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
index a3a8d17..30858b2 100644
--- a/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
@@ -656,6 +656,32 @@ xgbe_serdes: phy@231e000 {
 		rx-force	= <1 1>;
 		tx-coeff	= <2 0 0 12 4>; /* c1 c2 cm att vreg */
 	};
+
+	mcu-firmware {
+		status = "disabled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		lane@0 {
+			status = "disabled";
+			reg = <0>;
+			rate = <0>; /* 0 = default, 1 = 1g, 2 = 10g */
+			autonegotiate;
+			/*tx_pause;*/
+			/*rx_pause;*/
+			10g_train;
+			/*fec;*/
+		};
+		lane@1 {
+			status = "disabled";
+			reg = <1>;
+			rate = <0>; /* 0 = default, 1 = 1g, 2 = 10g */
+			autonegotiate;
+			/*tx_pause;*/
+			/*rx_pause;*/
+			10g_train;
+			/*fec;*/
+		};
+	};
 };
 
 netcpx: netcp@2f00000 {
diff --git a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
index 6abebb0..e78fb74 100644
--- a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
@@ -595,6 +595,32 @@ xgbe_serdes: phy@231e000 {
 		rx-force	= <1 1>;
 		tx-coeff	= <2 0 0 12 4>; /* c1 c2 cm att vreg */
 	};
+
+	mcu-firmware {
+		status = "disabled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		lane@0 {
+			status = "disabled";
+			reg = <0>;
+			rate = <0>; /* 0 = default, 1 = 1g, 2 = 10g */
+			autonegotiate;
+			/*tx_pause;*/
+			/*rx_pause;*/
+			10g_train;
+			/*fec;*/
+		};
+		lane@1 {
+			status = "disabled";
+			reg = <1>;
+			rate = <0>; /* 0 = default, 1 = 1g, 2 = 10g */
+			autonegotiate;
+			/*tx_pause;*/
+			/*rx_pause;*/
+			10g_train;
+			/*fec;*/
+		};
+	};
 };
 
 netcpx: netcp@2f00000 {
-- 
1.9.1

